Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -426,6 +427,31 @@ void ImageEditor::set_active_layer(Layer* layer)
     }
 }
 
+ErrorOr<void> ImageEditor::add_new_layer_from_selection()
+{
+    auto current_layer_selection = image().selection();
+    if (current_layer_selection.is_empty())
+        return Error::from_string_literal("There is no active selection to create layer from.");
+
+    // save offsets of selection so we know where to place the new layer
+    auto selection_offset_x = current_layer_selection.bounding_rect().x();
+    auto selection_offset_y = current_layer_selection.bounding_rect().y();
```

Comment:
Since you only use these to construct a new IntPoint, you could use:
```suggestion
    auto selection_offset = current_layer_selection.bounding_rect().location();
```

Reply:
fixed!